### ISSUE TITLE
whitespace rule needs to be broken apart?

### ISSUE THREAD

**Issue by axefrog** [palantir/tslint#2327]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
let foo = 0; // MUST have a space around the equals
if(foo) { // MUST NOT have a space after the `if` keyword
}
```

with `tslint.json` configuration:

```json
{
    "whitespace": [
      true,
      "check-decl",
      "check-separator",
      "check-type"
    ]
}
```

I am sure I must be missing something because it seems obvious that people aren't going to want to apply a uniform `true` or `false` to all spacing scenarios, especially ones that are unrelated such as the space around an assignment operator, versus a space after an `if` keyword.

How do I enforce spacing in certain case, and enforce the _absence_ of spaces in other scenarios?

**Comment by ajafff** [comment]
Currently there is not way to do what you want with the existing rules.

I agree that the options for `whitespace` need an overhaul. Instead of string options that should be an object with boolean values.

As a bonus we could merge `typedef-whitespace` and `space-before-function-paren` into this rule.

**Comment by JoshuaKGoldberg** [comment]
Agreed. We should see a more detailed discussion of what to do here before taking action.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
